mmeval/metrics/instance_seg.py

```diff
@@ -0,0 +1,288 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import numpy as np
+import warnings
+from copy import deepcopy
+from typing import Dict, List, Optional, Sequence
+
+from mmeval.core.base_metric import BaseMetric
+from mmeval.metrics._vendor.scannet import scannet_eval
+
+
+class InstanceSeg(BaseMetric):
+    """3D instance segmentation evaluation metric.
+
+    Args:
+        dataset_meta (dict, optional): Provide dataset meta information.
+        classes (List[str], optional): Provide dataset classes information as
+            an alternative to dataset_meta.
+        valid_class_ids (List[int], optional): Provide dataset valid class ids
+            information as an alternative to dataset_meta.
+
+    Example:
+        >>> import numpy as np
+        >>> from mmeval import InstanceSegMetric
+        >>> seg_valid_class_ids = (3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 14, 16, 24,
+        >>>                 28, 33, 34, 36, 39)
+        >>> class_labels = ('cabinet', 'bed', 'chair', 'sofa', 'table', 'door',
+        ...         'window', 'bookshelf', 'picture', 'counter', 'desk',
+        ...         'curtain', 'refrigerator', 'showercurtrain', 'toilet',
+        ...         'sink', 'bathtub', 'garbagebin')
+        >>> dataset_meta = dict(
+        ...     seg_valid_class_ids=seg_valid_class_ids, classes=class_labels)
+        >>>
```

Comment:
The example is too complicated and needs to be simplified~

Reply:
It is hard to simplify it for this metric need many code to generate input.